src/main/ngapp/src/app/dataservices/shared/dataservice.service.ts

```diff
@@ -358,40 +374,122 @@ export class DataserviceService extends ApiService {
+  }
+
+  /**
+   * Download a dataservice as a jar archive
    * @param {string} dataserviceName the dataservice name
    * @returns {Observable<boolean>}
    */
-  public exportDataservice(dataserviceName: string): Observable<boolean> {
-    const repoPathKey = this.appSettings.GIT_REPO_PATH_KEY;
-    const repoBranchKey = this.appSettings.GIT_REPO_BRANCH_KEY;
-    const repoUsernameKey = this.appSettings.GIT_REPO_USERNAME_KEY;
-    const repoPasswordKey = this.appSettings.GIT_REPO_PASSWORD_KEY;
-    const repoAuthorEmailKey = this.appSettings.GIT_REPO_AUTHOR_EMAIL_KEY;
-    const repoAuthorNameKey = this.appSettings.GIT_REPO_AUTHOR_NAME_KEY;
-    const repoFilePathKey = this.appSettings.GIT_REPO_FILE_PATH_KEY;
-
+  public downloadDataservice(dataserviceName: string): Observable<boolean> {
     // The payload for the rest call
     const payload = {
-      "storageType": "git",
-      "dataPath": "/" + this.getKomodoUserWorkspacePath() + "/" + dataserviceName,
-      "parameters":
-        {
-          [repoPathKey] : this.appSettings.getGitRepoProperty(repoPathKey),
-          [repoBranchKey] : this.appSettings.getGitRepoProperty(repoBranchKey),
-          [repoFilePathKey] : dataserviceName,
-          [repoUsernameKey] : this.appSettings.getGitRepoProperty(repoUsernameKey),
-          [repoPasswordKey] : btoa(this.appSettings.getGitRepoProperty(repoPasswordKey)),
-          [repoAuthorNameKey] : this.appSettings.getGitRepoProperty(repoAuthorNameKey),
-          [repoAuthorEmailKey] : this.appSettings.getGitRepoProperty(repoAuthorEmailKey)
-        }
+      "storageType": "file",
+      "dataPath": this.getKomodoUserWorkspacePath() + "/" + dataserviceName,
+      "parameters": {}
     };
 
     const url = environment.komodoImportExportUrl + "/" + DataservicesConstants.dataservicesExport;
 
     return this.http
       .post(url, payload, this.getAuthRequestOptions())
       .map((response) => {
+        let status = response.json();
+
+        if (! status.downloadable) {
+          throw new Error(payload.dataPath + " is not downloadable");
+        }
+
+        if (! status.content) {
+          throw new Error(payload.dataPath + " has no content");
+        }
+
+        const name = status.Name || dataserviceName;
+        const fileType = status.type || 'data';
+        const enc = status.content;
+
+        const contentType = fileType === "zip" ? 'application/zip' : 'text/plain;charset=utf-8';
+        const dataBlob = this.b64toBlob(enc, contentType);
+
+        const fileExt = ( fileType == "-vdb.xml" || fileType == "-connection.xml" ) ? fileType : "." + fileType;
+
+        saveAs(dataBlob, name + fileExt);
+
+        return response.ok;
+      })
+      .catch( ( error ) => this.handleError( error ) );
+  }
+
+  /**
+   * Publish a dataservice
+   * @param {string} dataserviceName the dataservice name
+   * @returns {Observable<boolean>}
+   */
+  public publishDataservice(dataserviceName: string): Observable<boolean> {
```

Comment:
Would be nice if this could also be implemented in `MockDataserviceService` so that the non-OpenShift dev mode would work.